scripts/create_changelog.py

```diff
@@ -0,0 +1,329 @@
+            'You can create one at https://github.com/settings/tokens: '))
+
+    if personal_access_token is None:
+        python_utils.PRINT(
+            'No personal access token provided, please set up a personal '
+            'access token at https://github.com/settings/tokens and re-run '
+            'the script')
+        return
+    g = github.Github(personal_access_token)
+    repo_fork = g.get_repo('%s/oppia' % github_username)
+
+    while True:
+        python_utils.PRINT(
+            '******************************************************')
+        python_utils.PRINT(
+            'Please update %s to:\n- have a correct changelog for '
+            'updating the CHANGELOG file\n- have a correct list of new '
+            'authors and contributors to update AUTHORS, CONTRIBUTORS '
+            'and developer_names section in about-page.directive.html\n'
+            'Confirm once you are done by entering y/ye/yes.\n' % (
+                feconf.RELEASE_SUMMARY_FILEPATH))
+        answer = python_utils.INPUT().lower()
+        if answer in ['y', 'ye', 'yes']:
+            break
+
+    current_release_version = branch_name[len(
+        common.RELEASE_BRANCH_NAME_PREFIX):]
+    source_branch = 'develop'
+    target_branch = 'update-changelog-for-releasev%s' % current_release_version
+    release_summary_lines = []
+    # This complete code block is wrapped in try except since in case of
+    # an exception, we should ensure that the changes made to AUTHORS,
+    # CONTRIBUTORS, CHANGELOG and about-page are reverted and the branch
+    # created is deleted. This is required to avoid errors when running
+    # the script again.
+    try:
+        with python_utils.open_file(
+            feconf.RELEASE_SUMMARY_FILEPATH, 'r') as release_summary_file:
+            release_summary_lines = release_summary_file.readlines()
+
+        update_changelog(
+            release_summary_lines, current_release_version)
+        update_authors(release_summary_lines)
+        update_contributors(release_summary_lines)
+        update_developer_names(release_summary_lines)
+
+        python_utils.PRINT(
+            'Creating new branch with updates to AUTHORS, CONTRIBUTORS, '
+            'CHANGELOG and about-page...')
+        sb = repo_fork.get_branch(source_branch)
+        repo_fork.create_git_ref(
+            ref='refs/heads/%s' % target_branch, sha=sb.commit.sha)
+
+        for filepath in [
+                ABOUT_PAGE_FILEPATH, CONTRIBUTORS_FILEPATH, AUTHORS_FILEPATH,
+                CHANGELOG_FILEPATH]:
+            contents = repo_fork.get_liness(filepath, ref=target_branch)
+            with python_utils.open_file(filepath, 'r') as f:
+                repo_fork.update_file(
+                    contents.path, 'Update %s' % filepath, f.read(),
+                    contents.sha, branch=target_branch)
+        common.run_cmd(GIT_CMD_TEMPLATE_CHECKOUT % ('%s %s %s %s') % (
```

Comment:
I think you need to add an interactive step where the author can review the diff, make changes if necessary, and then say "yes, this is correct, please go ahead".

Reply:
Done.